bootstrap/kubeadm/api/v1alpha3/kubeadmbootstrapconfig_types.go

```diff
@@ -67,7 +75,15 @@ type KubeadmConfigStatus struct {
 	// Ready indicates the BootstrapData field is ready to be consumed
 	Ready bool `json:"ready,omitempty"`
 
-	// BootstrapData will be a cloud-init script for now
+	// DataSecretName is the name of the secret where the output data is stored.
+	// +optional
+	DataSecretName string `json:"dataSecretName,omitempty"`
```

Comment:
Why not an object reference here like we do for other refs? We could add validation to ensure the reference is to a Secret vs other types.

Reply:
A name is fine here.

Reply:
Validation could happen here as well, I figured given that secrets are primitives and we always want to be in the same namespace we can use just names as references. 

I can change if you all prefer

Reply:
The string should be fine, but should we add some validation to ensure that it only accepts valid [Kubernetes names](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)?

Reply:
Sure

Reply:
Is there a regex I can use somewhere?

Reply:
Something like `[0-9az-\.]){1,253}`?

Untested, and I could have b0rked the syntax/escaping.

Reply:
I believe this is what you ultimately want: https://github.com/kubernetes/apimachinery/blob/adc6f4cd9e7d28fc1c1e5efc658d940d55c0f356/pkg/util/validation/validation.go#L155-L174

Reply:
If you do it in the webhook, calling `IsDNS1123Subdomain` is what you want.